File: MergingCommunities/MergingCommunities.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> parent, population;

int find(int i) {
    if (i != parent[i]) {
        parent[i] = find(parent[i]);
    }
    return parent[i];
}

void Union(int i, int j) {
    int i_id = find(i);
    int j_id = find(j);
    if (i_id != j_id) {
        if (population[i_id] > population[j_id]) {
            parent[j_id] = i_id;
            population[i_id] += population[j_id];
        } else {
            parent[i_id] = j_id;
            population[j_id] += population[i_id];
        }
    }
}

int main() {
    int n, q;
    char type;
    cin >> n >> q;
    parent.resize(n+1);
    population.resize(n+1);
    for (int i = 1; i <= n; i++) {
        parent[i] = i;
        population[i] = 1;
    }
    while (q--) {
        cin >> type;
        if (type == 'M') {
            int i, j;
            cin >> i >> j;
            Union(i, j);
        } else {
            int i;
            cin >> i;
            cout << population[find(i)] << endl;
        }
    }
    return 0;
}

File: MergingCommunities/MergingCommunities.py
def find(i, parent):
    if i != parent[i]:
        parent[i] = find(parent[i], parent)
    return parent[i]

def union(i, j, parent, size):
    i_id = find(i, parent)
    j_id = find(j, parent)
    if i_id != j_id:
        if size[i_id] > size[j_id]:
            parent[j_id] = i_id
            size[i_id] += size[j_id]
        else:
            parent[i_id] = j_id
            size[j_id] += size[i_id]

n, q = map(int, input().split())
parent = list(range(n+1))
size = [1]*(n+1)
for _ in range(q):
    query = input().split()
    if query[0] == 'M':
        union(int(query[1]), int(query[2]), parent, size)
    else:
        print(size[find(int(query[1]), parent)])
            
